master/buildbot/www/resource.py

```diff
@@ -117,3 +119,19 @@ def render(self, request):
         redir = self.base_url + self.basepath
         request.redirect(redir)
         return redir
+
+
+class InternalResource(Resource):
+    """
+    Resource available only for authorized users if authorization is enabled.
+    """
+    def render(self, request):
```

Comment:
I don't think it is needed to factorize an InternalResource if this is not needed. (principle of Minimum Viable Product)

Reply:
so you can just implment this in the render function of IndexResource